src/caesium/binding.clj

```diff
@@ -6,11 +6,15 @@
   namespaces."
   (:require [clojure.string :as s]
             [clojure.math.combinatorics :as c]
-            [medley.core :as m]
-            [clojure.string :as str])
+            [medley.core :as m])
   (:import [jnr.ffi LibraryLoader LibraryOption]
-           [jnr.ffi.annotations In Out Pinned LongLong]
-           [jnr.ffi.types size_t]))
+           [jnr.ffi.annotations
+            #_{:clj-kondo/ignore [:unused-import]} In
+            #_{:clj-kondo/ignore [:unused-import]} Out
+            #_{:clj-kondo/ignore [:unused-import]} Pinned
+            #_{:clj-kondo/ignore [:unused-import]} LongLong]
```

Comment:
This should be a clj-kondo bug: these _are_ used.

Reply:
Yeah, they are. I added the directives to make clj-kondo happy. I should probably revert that—that's a weird fix, you're right.

Skimming through the clj-kondo issues revealed https://github.com/borkdude/clj-kondo/issues/911, which closely resembles the issue we have here. But the latest clj-kondo still gives the same unused import warnings.

I'll file an issue for our case in clj-kondo. Please stay tuned :)

Reply:
Well, I just gave it a second thought trying to come up with an example illustrating the issue to submit to clj-kondo, and I reckon the errors reported are actually fine, given the way things are implemented in caesium and the way clj-kondo handles macro expansions.

First of all, clj-kondo only expands a small set of macros as outlined here: https://cljdoc.org/d/clj-kondo/clj-kondo/2020.09.09/doc/configuration#unrecognized-macros

And then, all of the `In`, `Out`, `Pinned`, and `LongLong` usages happen to be in a _quoted_ list in caesium. In such form they are _not_ bound to the imported symbols until they are later expanded in the `defsodium` macro. 

So, clj-kondo, albeit being pretty naïve here, correctly complains the imports are not used.

I'm not sure what to do with this. Any ideas?

Reply:
Hm, OK, I think it's still a bug but I see your point that it's unreasonable to expect clj-kondo to fix this especially on short timeframes. I think this is good to go as is then?